mintapi/api.py

```diff
@@ -616,75 +616,23 @@ def populate_extended_account_detail(self, accounts):  # {{{
-            if isinstance(hist, str) or hist > 12:
-                hist = 12  # MINT_ROOT_URL only calls last 12 months of budget data
-            elif hist < 1:
-                hist = 1
-
-            bgt_cur_mo = max(map(int, income.keys()))
-            min_mo_hist = bgt_cur_mo - hist
-
-            # Initialize and populate dictionary for return
-            #   Output 'budgets' dictionary with structure
-            #       { "YYYYMM": {"spending": [{"key": value, ...}, ...],
-            #                      "income": [{"key": value, ...}, ...] } }
-            budgets = {}
-            for months in range(bgt_cur_mo, min_mo_hist, -1):
-                budgets[mos_to_yrmo(months)] = {}
-                budgets[mos_to_yrmo(months)]["income"] = income[str(months)]["bu"]
-                budgets[mos_to_yrmo(months)]["spending"] = spending[str(months)]["bu"]
-
-            # Fill in the return structure
-            for month in budgets.keys():
-                for direction in budgets[month]:
-                    for budget in budgets[month][direction]:
-                        budget = self.__format_budget_categories(budget, categories)
-
+    def get_budgets(self):
+        budgets = self.__call_budgets_endpoint()
+        if "Budget" in budgets.keys():
+            for i in budgets["Budget"]:
+                i["lastUpdatedDate"] = i["metaData"]["lastUpdatedDate"]
+                i.pop("metaData", None)
         else:
-            # Make the skeleton return structure
-            budgets = {
-                "income": income[str(max(map(int, income.keys())))]["bu"],
-                "spend": spending[str(max(map(int, spending.keys())))]["bu"],
-            }
-
-            # Fill in the return structure
-            for direction in budgets.keys():
-                for budget in budgets[direction]:
-                    budget = self.__format_budget_categories(budget, categories)
-
-        return budgets
+            raise MintException("Cannot find budget data")
+        return budgets["Budget"]
 
     def __call_budgets_endpoint(self):
-        # Issue request for budget utilization
-        first_of_this_month = date.today().replace(day=1)
-        eleven_months_ago = (first_of_this_month - timedelta(days=330)).replace(day=1)
-        url = "{}/getBudget.xevent".format(MINT_ROOT_URL)
-        params = {
-            "startDate": convert_date_to_string(eleven_months_ago),
-            "endDate": convert_date_to_string(first_of_this_month),
-            "rnd": Mint.get_rnd(),
-        }
-        return json.loads(self.get(url, params=params, headers=JSON_HEADER).text)
-
-    def __format_budget_categories(self, budget, categories):
-        category = self.get_category_object_from_id(budget["cat"], categories)
-        budget["cat"] = category["name"]
-        parent = self._find_parent_from_category(category, categories)
-        budget["parent"] = parent["name"]
-        return budget
+        return self.get(
```

Comment:
Is this pattern used elsewhere? I'm curious if we should have some kind get_json_with_headers if so. If not, no worries.

Reply:
@mrooney Yeah this will be a pattern that will be very prevalent in the new endpoint changes.  Once this merges, I can do some cleanup where I create a generic method to better encompass this functionality.